### ISSUE TITLE
init should be safer and abort if no --datadir is found

### ISSUE THREAD

**Issue by chevdor** [ethereum/go-ethereum#2552]
#### System information

Geth
Version: 1.5.0-unstable-2d7d7ef2
Protocol Versions: [63 62 61]
Network Id: 1
Go Version: go1.6.2
OS: darwin
GOPATH=
GOROOT=/usr/local/Cellar/go/1.6.2/libexec

OS & Version: OSX
Commit hash : (if `develop`)

#### Expected behaviour
geth init is safe enough to prevent  corrupting the main blockchain.

#### Actual behaviour
There are 2 issues:
* if --datadir is not provided, it defaults to the mainnet
* if --datadir is provided *after* the init command, it defaults to mainnet

#### Steps to reproduce the behaviour

All of the following will damage your mainnet blockchain, beware! While this is understandable for the first command, for the 2 others, you probably don´t expect your mainnet to get touched.
* geth init genesis.json
* geth -datadir /Users/chevdor/Library/Ethereum/privatenet/ init genesis.json (notice the error, -datadir instead of --datadir)
* geth init genesis.json --datadir /Users/chevdor/Library/Ethereum/privatenet/

**Comment by karalabe** [comment]
Alternatively a better approach imho would be to fail if the data directory already exists or require the user to specify `--force`.

**Comment by chevdor** [comment]
I totally agree. I would rather have to type a little more to be sure that destroy my data, as I did today because I typed -datadir instead of --datadir 😢

**Comment by karalabe** [comment]
The two are interchangeable btw ;)

**Comment by chevdor** [comment]
That would be great. I find the:

    geth init genesis.json --datadir /Users/chevdor/Library/Ethereum/privatenet/

very tricky as it 'looks' good but geth totally ignores --datadir as it comes in the wrong position (guess).

**Comment by obscuren** [comment]
For the record, you don't overwrite any chain data. When you made a mistake or forgot the `--datadir` the only thing that happens is that you tell geth "block 0 (i.e. genesis) = [this block]". You can do this as many times as you want and hence it allows you to revert back to any previous genesis block.

**Comment by chevdor** [comment]
Thanks for the clarification @obscuren, this is indeed good to know to revert a mistake.
However, it sounds like it would put the env. in a weird state where the data in the blockchain no longer matches the genesis block.

Additionally, for the documentation purpose, would you mind providing the commands allowing resetting the genesis block to the proper one for both testnet and mainnet?
